dotcom-rendering/src/web/components/ShareIcons.tsx

```diff
@@ -123,15 +125,32 @@ const iconStyles = ({
 	}
 `;
 
-const encodeUrl = (pageId: string): string => {
-	return encodeURIComponent(`https://www.theguardian.com/${pageId}`);
-};
+const getUrl = ({
+	pageId,
+	CMP,
+	blockId,
+}: {
+	pageId: string;
+	CMP?: string;
+	blockId?: string;
+}) => {
+	const searchParams = new URLSearchParams({});
+	if (CMP) searchParams.append('CMP', CMP);
+	if (blockId) searchParams.append('page', `with:block-${blockId}`);
 
-const encodeTitle = (webTitle: string): string => {
-	return encodeURIComponent(webTitle.replace(/Leave.EU/gi, 'Leave. EU'));
```

Comment:
Does `URL` automatically encode?

Reply:
`URLSearchParams` does, yes: [see for yourself](https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgQwE4ILYRgXhmApgdxgFUAlAGQGVdlgALABWTQgAoBvAWACgZeYoAllAA2uAFwwARAEEYASTBRcSXNAFgA5jApQk6jRIAHMYQhgAzFGADnCARFwBCGAHJ8tAJ44QUAPwuYAB4AIyQYAHoAPkluAF8ASm5uUEgQUQA6YRANFkQUdHSoEB09TRZ4xK5w8JgAbTIAeQBxAF0JSUERXExpAGoFJRU1TV6S-QBSAGY+8YBhSfHpAAZe017LBDA5hekAdjsHcYAmAEZe4933D16wHymAMQvzydnQ86PHyYBRSRhuIA).

Reply:
I’ve done some testing and it works well on Twitter and Mail, but there is a small risk that `+` for spaces instead of `%20` makes the social platforms unhappy.